concourse/pipelines/modelbase.py

```diff
@@ -69,10 +70,28 @@ def custom_init(self, raw_dict: dict):
         if raw_dict.get('output_dir', None):
             name = raw_dict['output_dir']
             self.add_output(name + '_path', name + '_path')
+        # normalise step-arguments, if given. For convenience we allow both primitive integers and strings
+        # and lists containing entries of those two types.
+        args = raw_dict.get('arguments', None)
+        if args:
+            if isinstance(args, (str, int)):
```

Comment:
do this the other way around: check if it is a list; that way, you do not have to restrict the allowed types to str and int
also: there is a third case: it could be a dict. raise a ModelValidationError in this case